### PR TITLE
fix multimodal hash feature

### PR DESCRIPTION
**Pull request by huangtingwei9988**
## Motivation

When a hash calculation is performed on a tensor, it is usually calculated based on the tensor's address. When two tensors have the same value but different addresses, their hash values ​​are different. 
```
# example

import torch

randn_tensor = torch.randn(4, 1024)
print(hash(randn_tensor))
print(hash(randn_tensor.clone()))

# result 
# 139792403318000
# 139786750322192

```
For sglang's [hash_feature](https://github.com/sgl-project/sglang/blob/ca8d02abd520194bc0d182d62aeb39d9496ca91d/python/sglang/srt/managers/schedule_batch.py#L204) 
The same multimodal input will get different hash values. This causes the multi-modal radix cache  fail to hit
```

# example

import torch
tensor_list_1 = []
for i in range(10):
    tensor_list_1.append(torch.randn(4, 1024))
print(hash(tuple(tensor_list_1)))

tensor_list_2 = []
for i in range(10):
    tensor_list_2.append(tensor_list_1[i].clone())
print(hash(tuple(tensor_list_2)))


# result 
# -2006800841837242759
# 7092966304206677122
```

## Modifications

- [x] fix bug

## Checklist

- [ ] Format your code according to the [Code Formatting with Pre-Commit](https://docs.sglang.ai/references/contribution_guide.html#code-formatting-with-pre-commit).
- [ ] Add unit tests as outlined in the [Running Unit Tests](https://docs.sglang.ai/references/contribution_guide.html#running-unit-tests-adding-to-ci).
- [ ] Update documentation / docstrings / example tutorials as needed, according to [Writing Documentation](https://docs.sglang.ai/references/contribution_guide.html#writing-documentation-running-docs-ci).
- [ ] Provide throughput / latency benchmark results and accuracy evaluation results as needed, according to [Benchmark and Profiling](https://docs.sglang.ai/references/benchmark_and_profiling.html) and [Accuracy Results](https://docs.sglang.ai/references/accuracy_evaluation.html).
- [ ] For reviewers: If you haven't made any contributions to this PR and are only assisting with merging the main branch, please remove yourself as a co-author when merging the PR.
- [ ] Please feel free to join our Slack channel at https://slack.sglang.ai to discuss your PR.

### CODE REVIEW

**Comment by zhyncs** [comment]
Thanks! @huangtingwei9988 @AniZpZ

**Comment by mickqian** [comment]
LGTM. Thanks!
